### PR TITLE
Spring security implemented

### PR DESCRIPTION
**Pull request by hpavlo**

### CODE REVIEW

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
```
Comment (YuriiHryniv) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:12]: Apparently, you forgot to add this class to WebAppInitializer.
image

**Review by YuriiHryniv — CHANGES_REQUESTED** [review]

**Review by YuriiHryniv — APPROVED** [review]
LGTM!

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        auth.inMemoryAuthentication()
+                .withUser("bob@user.com")
+                .password(getEncoder().encode("1234"))
+                .roles("USER");
+    }
+
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .anyRequest().authenticated()
```
Comment (hronograf) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:24]: let's add `@Override` annotation
also looks like users can't register themselves

File: src/main/java/mate/academy/spring/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -4,24 +4,23 @@
 import mate.academy.spring.dao.UserDao;
 import mate.academy.spring.model.User;
 import mate.academy.spring.service.UserService;
-import mate.academy.spring.util.HashUtil;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
     private static final int SALT_LENGTH = 10;
```
Comment (hronograf) [inline src/main/java/mate/academy/spring/service/impl/UserServiceImpl.java:12]: do we need `SALT_LENGTH` constant?

**Review by hronograf — CHANGES_REQUESTED** [review]

**Review by hronograf — APPROVED** [review]

**Pull request closed without merging**